articles/service-fabric/service-fabric-reliable-services-communication-remoting.md

```diff
@@ -311,50 +301,59 @@ Following example uses Json Serialization with Remoting V2.
             var list = new List<ArraySegment<byte>> { segment };
             return new OutgoingMessageBody(list);
         }
-
+    
         public IServiceRemotingResponseMessageBody Deserialize(IncomingMessageBody messageBody)
         {
             using (var sr = new StreamReader(messageBody.GetReceivedBuffer()))
-
             using (var reader = new JsonTextReader(sr))
             {
                 var serializer = JsonSerializer.Create(new JsonSerializerSettings()
                 {
                     TypeNameHandling = TypeNameHandling.All
                 });
-
+    
                 return serializer.Deserialize<JsonRemotingResponseBody>(reader);
             }
         }
     }
-
-    internal class JsonRemotingResponseBody : IServiceRemotingResponseMessageBody
+    
+    class JsonRemotingResponseBody : IServiceRemotingResponseMessageBody
     {
-        public object Value;
-
+        public string Value { get; set; }
+    
         public void Set(object response)
         {
-            this.Value = response;
+            Value = JsonConvert.SerializeObject(response);
```

Comment:
@brutaldev 
Wont this means double serialization and hence less performant ?  For the types like int , guid,, date time where json doesn't deserialize it properly , you could add custom converter for those.

Reply:
I don't see the double serialization here, perhaps you meant extra/unnecessary serialization? The `response` object could be a regular type (string, int, long, DateTime etc) or it could be a much more complex custom data type. To improve performance  the type could be checked before going through the serializer/deserializer and avoid if it's cant make the trip without it.

The issue here is that complex types containing things like `long` data types incorrectly get serialized as int and then break on the other side with data type conversion errors.